internal/lib/lib.go

```diff
@@ -2150,37 +2153,43 @@ func IsChanClosed(ch <-chan struct{}) bool {
 func GetIPFromNode(node *v1.Node) (string, string) {
 	var nodeV4, nodeV6 string
 	nodeAddrs := node.Status.Addresses
-	for _, addr := range nodeAddrs {
-		if addr.Type == corev1.NodeInternalIP {
-			nodeIP := addr.Address
-			if utils.IsV4(nodeIP) {
-				//Use first IP in the list
-				if nodeV4 == "" {
-					nodeV4 = nodeIP
-				}
-			} else if nodeV6 == "" {
-				nodeV6 = nodeIP
-			}
-		}
-	}
-	if GetIPFamily() == "V6" {
-		if GetCNIPlugin() == CALICO_CNI {
+
+	v4enabled := GetIPFamily() == "V4" || GetIPFamily() == "V4_V6"
```

Comment:
can we optimize it by calling GetIPFamily once?

Reply:
updated